assignments/objects.js

```diff
@@ -18,31 +18,67 @@ const example = {
 }
 
 // Write your intern objects here:
+const intern1 = {
+  "id": 1,
+  "email":"mmelloy0@psu.edu",
+  "name": "Mitzi",
+  "gender":"F"
+}
+const intern2 = {
+  "id": 2,
+  "email":"kdiben1@tinypic.com",
+  "name":"Kennan",
+  "gender":"M"
+}
+intern2.speak = () => {
+  console.log(`Hello , my name is ${intern2.name}`)};
```

Comment:
Logging `intern2`'s name is a little brittle. Meaning, what about if the variable name was different? What about if the object is inside an array of intern objects? In those situations, the name in the object is still "Kennan", but the `speak` method won't work because it depends on the "Kennan" object to be called `intern2`.

You could just hardcode Kennan's name in the `console.log` to get around this. You can also use the `this` keyword, though you _cannot_ use the arrow function to do that:

```js
intern2.speak = function() {
  console.log(`Hello , my name is ${this.name}`)}; // Hello , my name is Kennan
```

❗️ Arrow Function:

```js
intern2.speak = () => {
  console.log(`Hello , my name is ${this.name}`)}; // Hello , my name is undefined
```

We'll learn more about the `this` keyword in future lessons.